### ISSUE TITLE
ANSI: SELECT ... FOR UPDATE not supported

### ISSUE THREAD

**Issue by Yiwen-Gao** [sqlfluff/sqlfluff#4205]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I have a SELECT statement that retrieves the oldest element from a table, but sqlfluff is unable to lint it correctly.
```sql
SELECT job
FROM queue
ORDER BY last_update_time ASC
FOR UPDATE SKIP LOCKED
LIMIT 1;
```

### Expected Behaviour

I think sqlfluff should support the `FOR UPDATE` clause because I believe it's part of the [ANSI SQL standard](https://cockroachlabs.com/blog/select-for-update/#what-is-select-for-update).

### Observed Behaviour

When I try to lint, sqlfluff fails to parse `FOR UPDATE` correctly.
```
L: 209 | P:   1 |  PRS | Line 209, Position 1: Found unparsable section: 'FOR
                       | UPDATE SKIP LOCKED'
```

### How to reproduce

- Use any SELECT query with a FOR UPDATE clause
- Lint the query with sqlfluff

### Dialect

Postgres

### Version

sqlfluff: 1.4.5
Python: 3.8

### Configuration

[sqlfluff]
dialect=postgres
exclude_rules=
templater=placeholder
large_file_skip_byte_limit=0

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by Yiwen-Gao** [comment]
I see there's a `ForClauseSegment` for MySQL--would it be better to add it for Postgres or all dialects?
https://github.com/sqlfluff/sqlfluff/blob/83636f24350483de31438441e050a8c54b61a187/src/sqlfluff/dialects/dialect_mysql.py#L1619-L1635
